Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.cpp

```diff
@@ -369,6 +369,8 @@ void AXIsolatedTree::queueRemovalsLocked(Vector<AXID>&& subtreeRemovals)
     ASSERT(m_changeLogLock.isLocked());
 
     m_pendingSubtreeRemovals.appendVector(WTFMove(subtreeRemovals));
+    if (m_protectedFromDeletionIDsIsDirty)
+        m_pendingProtectedFromDeletionIDs.formUnion(m_protectedFromDeletionIDs);
```

Comment:
> This line is just slightly confusing, do we really need most of the ids to end up in both sets?

Right now, yes — it follows the pattern of other `m_pending*` data structures we sync for isolated tree updates, and these specific two are required to make sure we don't delete objects from the accessibility thread data structures, nor the main-thread data structures (`AXIsolatedTree::m_nodeMap`) that help us make correct and efficient updates to the tree. But overall this whole system is over-complicated — I have ideas I'm pursuing this upcoming year to simplify things, so hopefully some of this can be removed outright.

> I notice elsewhere you used std::exchange to delete from m_protectedFromDeletionIDs and union with m_pendingProtectedFromDeletionIDs, and I just want to make sure that wouldn't make sense here.

Indeed, this is actually what we did prior to http://commits.webkit.org/281440@main. But it caused `m_protectedFromDeletionIDs` to be cleared too early, before a full tree-update cycle, causing the `m_nodeMap` to be updated incorrectly.